mule-extensions-api/src/main/java/org/mule/runtime/extension/api/notification/ExtensionNotification.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.notification;
+
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import org.mule.runtime.api.component.location.ComponentLocation;
+import org.mule.runtime.api.event.Event;
+import org.mule.runtime.api.notification.AbstractServerNotification;
+
+/**
+ * Represents notifications fired by an extension.
+ *
+ * @param <T> the type of data provided
+ * @since 1.1
+ */
+public class ExtensionNotification<T> extends AbstractServerNotification {
+
+  private static final long serialVersionUID = 6641419368045215517L;
+
+  private final Event event;
+  private final ComponentLocation componentLocation;
+  private final NotificationInfo<T> info;
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  ExtensionNotification(Event event, ComponentLocation componentLocation, NotificationInfo<T> info) {
+    super(null, info.getAction());
+    checkArgument(EXTENSION_EVENT_ACTION_START_RANGE < info.getAction(),
```

Comment:
it seems like this line of code is the only one which adds any value here, and you could easily do this in the notification emitter. I would like to see if it's possible to avoid the existence of this class, since it isn't really necessary for this API and increases the contact surface between the ext-api and the notifications api. I would like to be able to replace the notifications system without impacting the extensions